src/code/z_actor.c

```diff
@@ -5644,107 +5672,138 @@ s32 func_800374E0(PlayState* play, Actor* actor, u16 textId) {
                 } else {
-                    func_80035B18(play, actor, 0x301E);
+                    Actor_LoadResponseTextID(play, actor, 0x301E);
                 }
             }
             ret = 0;
             break;
         case 0x301E:
-            func_80035B18(play, actor, 0x3020);
+            Actor_LoadResponseTextID(play, actor, 0x3020);
             ret = 0;
             break;
         case 0x400C:
             if (msgCtx->choiceIndex == 0) {
-                func_80035B18(play, actor, 0x400D);
+                Actor_LoadResponseTextID(play, actor, 0x400D);
             }
             if (msgCtx->choiceIndex == 1) {
-                func_80035B18(play, actor, 0x400E);
+                Actor_LoadResponseTextID(play, actor, 0x400E);
             }
             ret = 0;
             break;
         case 0x7007:
-            func_80035B18(play, actor, 0x703E);
+            Actor_LoadResponseTextID(play, actor, 0x703E);
             ret = 0;
             break;
         case 0x703E:
-            func_80035B18(play, actor, 0x703F);
+            Actor_LoadResponseTextID(play, actor, 0x703F);
             ret = 0;
             break;
         case 0x703F:
-            func_80035B18(play, actor, 0x7042);
+            Actor_LoadResponseTextID(play, actor, 0x7042);
             ret = 0;
             break;
     }
 
     return ret;
 }
 
-u16 func_80037C30(PlayState* play, s16 arg1) {
-    return func_80035BFC(play, arg1);
+/**
+ * Returns a text ID value for an actor given the current set of world flags
+ * Note: Only used by Deku Tree Sprout, see an individual actor's overlay .c file for where their dialog logic actually resides
+ *
+ * @param play current game state
+ * @param actorID (s16) the actor we're trying to talk to
+ * @return (u16) text ID
+ */
+u16 Actor_GetTextID(PlayState* play, s16 actorID) {
+    return Actor_GetTextID32(play, actorID);
 }
 
-s32 func_80037C5C(PlayState* play, s16 arg1, u16 textId) {
-    func_80036E50(textId, arg1);
+/**
+ * Wrapper function w/ return type of s32
+ * Note: Only used by Deku Tree Sprout, see an individual actor's overlay .c file for where their dialog logic actually resides
+ * @see Actor_SetInfForTextID
+ */
+s32 Actor_SetInfForTextID32(PlayState* play, s16 actorID, u16 textId) {
+    Actor_SetInfForTextID(textId, actorID);
     return false;
 }
 
-s32 func_80037C94(PlayState* play, Actor* actor, s32 arg2) {
-    return func_800374E0(play, actor, actor->textId);
+/**
+ * Returns a text ID value for an actor given the current set of world flags
+ * Note: Only used by Deku Tree Sprout, see an individual actor's overlay .c file for where their dialog logic actually resides
+ *
+ * @param play current game state
+ * @param actor the actor we're trying to talk to
+ * @param pad (s32) unused
+ * @return 0 if we found and loaded a response, 1 if something has gone wrong
+ */
+s32 Actor_ProcessDialogChoice(PlayState* play, Actor* actor, s32 pad) {
```

Comment:
```suggestion
s32 Actor_ProcessDialogChoice(PlayState* play, Actor* actor, s32 unused) {
```
"pad" is used out of habit/semi-convention for variable declarations where it's needed to "pad" the stack for matching, but it doesn't make sense to use this name for arguments.
We use `unused` elsewhere